lib/Fmt_ast.ml

```diff
@@ -4182,30 +4241,44 @@ and fmt_signature_item c ?ext {ast= si; _} =
         $ hvbox_if (not box) 0 (fmt_item_extension c ctx ext)
         $ fmt_item_attributes c ~pre:(Break (1, 0)) atrs
         $ doc_after )
-  | Psig_include {pincl_mod; pincl_attributes; pincl_loc} ->
-      let pincl_attributes, isfunctor =
-        check_include_functor_attr pincl_attributes
-      in
+  | Psig_include
+      ({pincl_mod; pincl_attributes; pincl_loc; pincl_kind}, modalities) ->
       update_config_maybe_disabled c pincl_loc pincl_attributes
       @@ fun c ->
       let doc_before, doc_after, atrs =
-        let force_before = not (Mty.is_simple pincl_mod) in
+        let force_before =
+          (* CR modes: As a temporary hack, we force doc comments before
+             [include S @@ mode], because the parsetree doesn't currently
+             have attribute slots that support placing them after. *)
```

Comment:
I think we determined the problem is not really the lack of attribute slots, but just that the parser drops them in that position, so this comment is out of date.